clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/job/KubectlJobExecutor.java

```diff
@@ -151,6 +151,50 @@ public Void scale(KubernetesV2Credentials credentials, KubernetesKind kind, Stri
     return null;
   }
 
+  public List<Integer> historyRollout(KubernetesV2Credentials credentials, KubernetesKind kind, String namespace, String name) {
+    List<String> command = kubectlNamespacedAuthPrefix(credentials, namespace);
+
+    command.add("rollout");
+    command.add("history");
+    command.add(kind.toString() + "/" + name);
+
+    String jobId = jobExecutor.startJob(new JobRequest(command),
+        System.getenv(),
+        new ByteArrayInputStream(new byte[0]));
+
+    JobStatus status = backoffWait(jobId, credentials.isDebug());
+
+    if (status.getResult() != JobStatus.Result.SUCCESS) {
+      throw new KubectlException("Failed to get rollout history of " + kind + "/" + name + " from " + namespace + ": " + status.getStdErr());
+    }
+
+    String stdout = status.getStdOut();
+    if (StringUtils.isEmpty(stdout)) {
+      return new ArrayList<>();
+    }
+
+    // <resource> "name"
```

Comment:
This worries me just because I imagine that kubectl doesn't necessarily offer stability in its output. Why not parse JSON instead?

Reply:
There is no json output for this command :/ I'm hoping we'd catch regressions in our upcoming citests...